qt/component.h

```diff
@@ -253,7 +253,8 @@ class APPSTREAMQT_EXPORT Component
     void setMergeKind(MergeKind kind);
 
     QHash<QString, QString> custom() const;
-    QString customValue(const QString &key);
+    QString customValue(const QString &key) const;
+    QString customValue(const QString &key); //TODO Remove when we break ABI
```

Comment:
Hmm, can this be marked as `[[deprecated]]`, or will that cause trouble with the C++ compiler not being smart enough?

Reply:
No, added.

We'll just get deprecation warnings in cases where it's used without std::as_const. IIRC c++ prefers the non-const variant.

Reply:
> We'll just get deprecation warnings in cases where it's used without std::as_const. IIRC c++ prefers the non-const variant.

That's what I was worried about :-/

The benefit of the deprecation tag is that I will be able to easily spot and remove it once we break API though, a comment is harder to spot (especially when it doesn't contain the keyword "deprecated" ^^)

If there's deprecation warnings that can't be avoided, let me know and we can add a comment instead!